contracts/src/NativeTokenDestination.sol

```diff
@@ -239,21 +234,51 @@ contract NativeTokenDestination is
         return _isCollateralized();
     }
 
+    /**
+     * @dev See {IWrappedNativeToken-deposit}.
+     */
+    function deposit() public payable {
+        emit Deposit(msg.sender, msg.value);
+        _mint(msg.sender, msg.value);
+    }
+
+    /**
+     * @dev See {IWrappedNativeToken-withdraw}.
+     */
+    function withdraw(uint256 amount) public {
+        emit Withdrawal(msg.sender, amount);
+        _burn(msg.sender, amount);
+        payable(msg.sender).transfer(amount);
+    }
+
     /**
      * @dev See {INativeTokenDestination-totalSupply}.
+     * This implementation overrides the default ERC20 implementation. It provides the total
+     * supply of the native token, accounting for burned fees and tokens burnt to be transfered
+     * out to other chains. Native tokens transfered out to other chains are left locked in this
+     * contract forever, as the native minter precompile does not offer a "burnNativeCoin" interface.
      */
-    function totalSupply() external view returns (uint256) {
-        uint256 burned = BURNED_TX_FEES_ADDRESS.balance + token.balanceOf(BURN_FOR_TRANSFER_ADDRESS);
+    function totalSupply()
+        public
+        view
+        override (ERC20, IERC20, INativeTokenDestination)
+        returns (uint256)
+    {
+        uint256 burned = BURNED_TX_FEES_ADDRESS.balance + address(this).balance;
```

Comment:
will this burned amount include amounts that are briged and deposited?

Reply:
Oof, good point. Maybe we should rename this function `totalNativeAssetSupply`, and then add back the `totalSupply()` of the wrapped ERC20 token?

Reply:
hmm i need to think on this more, because since `NativeTokenDestination` is now an `IWrappedNativeToken` too, right now anyone can call `deposit()` on their own. That amount is included as burned too. I'm not sure what counts in the `totalSupply`, if it meant circulating supply of native tokens that isn't locked in this bridge contract, then it would make more sense right now.